pkg/operator/ceph/cluster/cluster.go

```diff
@@ -290,47 +291,37 @@ func clusterChanged(oldCluster, newCluster cephv1.ClusterSpec, clusterRef *clust
 	return false, ""
 }
 
-func (c *cluster) unsetOrchestrationRunning() {
-	c.orchRunMux.Lock()
-	c.orchestrationRunning = false
-	c.orchRunMux.Unlock()
-}
-
-func (c *cluster) setOrchestrationPending() {
-	c.orchPenMux.Lock()
-	c.orchestrationPending = true
-	c.orchPenMux.Unlock()
-}
-
 func (c *cluster) getOrchestrationRunning() bool {
-	c.orchRunMux.Lock()
-	defer c.orchRunMux.Unlock()
+	c.orchMux.Lock()
+	defer c.orchMux.Unlock()
 	return c.orchestrationRunning
 }
 
-func (c *cluster) getOrchestrationPending() bool {
-	c.orchPenMux.Lock()
-	defer c.orchPenMux.Unlock()
-	return c.orchestrationPending
+// unsetOrchestrationStatus resets the orchestration flags to its original status when
+// the orchestration is done
+func (c *cluster) unsetOrchestrationStatus() {
+	c.orchMux.Lock()
+	c.orchestrationRunning = false
+	c.orchestrationPending = true
```

Comment:
Why does completion of the orchestration running imply orchestration is pending?

Reply:
With `orchestrationPending` the `for-loop` in `createInstance` gets controlled. It is responsible to 
1) get into the loop (that is the reason why the variable needs to get initialized and reset to `true`) and
2) get the loop running as long as there is a new request pending.

Well, the name of this variable is somehow misleading. Any suggestions for a better name?

Reply:
So is the role more something like `readyForOrchestrationRun`?

Reply:
Well, not really. Because `orchestrationPending` set does not mean it is ready for orchestration (e.g. there is already an orchestration running).

Reply:
I think maybe my initial question didn't make sense. Why does the `unsetOrchestrationStatus` method, which I presume is called when an orchestration is finished running, set `orchestrationPending`? This seems like it will create a situation where orchestrations are infinitely pending, even when there are no updates. I would think that the only thing that should set `orchestrationPending = true` would be an event that triggers an orchestration run.

Reply:
`unsetOrchestrationStatus` gets called when `createInstance` is done -> the for-loop which does the actual orchestration does not get executed anymore. The next orchestration run will start as soon as  `createInstance` gets called e.g. if a node got added or the CRD got altered. Resetting ist needed `orchestrationPending` as this is one of the two criterias which allows to enter the for-loop (orchestration).
1) There is no other orchestration run currently (gets checked at the beginning of 'createInstance`)
2) `orchestrationPending` is set to true. This means that it has to be initialized as `true` and it needs to be reset to `true` when _all_ orchestration-runs are done otherwise the `for`-loop is not "enterable". Again, the name is misleading. But I have not better idea ;)